sql/core/v2.3/src/main/java/org/apache/spark/sql/execution/datasources/orc/OrcColumnVector.java

```diff
@@ -136,7 +138,9 @@ public int getInt(int rowId) {
   public long getLong(int rowId) {
     int index = getRowIndex(rowId);
     if (isTimestamp) {
-      return timestampData.time[index] * 1000 + timestampData.nanos[index] / 1000 % 1000;
+      Timestamp ts = new Timestamp(timestampData.time[index]);
+      ts.setNanos(timestampData.nanos[index]);
+      return DateTimeUtils.fromJavaTimestamp(ts);
```

Comment:
We can use a simpler API: `DateTimeUtils.fromJavaTimestamp(timestampData.asScratchTimestamp(index))`

Reply:
Under the hood, this re-uses the `Timestamp` object and should be more efficient.